contexts/AuthContext.tsx

```
@@ -0,0 +1,180 @@
+'use client';
+
+import React, { createContext, useContext, useMemo, useState, useEffect } from 'react';
+import { useAccount } from 'wagmi';
+import { usePrivy, useWallets } from '@privy-io/react-auth';
+import { adminAddress } from '@/lib/constant';
+
+interface AuthContextType {
+  address: string | null;
+  authenticated: boolean;
+  user: any;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)